### PR TITLE
fix: some bugs and some improvement in create new button

### PR DESCRIPTION
**Pull request by sahsisunny**
### Changes Made

- **Hamburger Menu Arrow Rotation:**
  - Updated the CSS and animation script to ensure the hamburger menu arrow rotates smoothly when the menu is opened.

- **Hamburger Icon Design:**
  - Replaced the odd five-line design of the hamburger icon with the arrow design for a more consistent and recognized look.

- **Plus Icon Visibility:**
  - Implemented conditional checks to hide or disable the plus icon specifically in the "create new page" context, improving the overall user experience.

- **Countdown Animation:**
  - Adjusted the styling of the countdown animation to create adequate spacing, preventing any visual annoyance by ensuring it no longer touches the surrounding text.

### Breaking Changes

- [ ] Yes
- [x] No

### Development Tested?

- [x] Yes
- [ ] No

### Under Feature Flag

- [ ] Yes
- [x] No

### Database Changes

- [ ] Yes
- [x] No

### Screenshots

https://github.com/Real-Dev-Squad/tiny-site-frontend/assets/70854507/b57002ec-8d58-4e1d-b854-3ddf7183c4a2

### Test Coverage

------------------------------|---------|----------|---------|---------|-------------------
File                          | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s
------------------------------|---------|----------|---------|---------|-------------------
All files                     |   96.98 |    89.02 |   92.77 |   97.94 |
 fixtures                     |     100 |      100 |     100 |     100 |
  users.ts                    |     100 |      100 |     100 |     100 |
 src/components/App           |     100 |      100 |     100 |     100 |
  InputSection.tsx            |     100 |      100 |     100 |     100 |
  OutputSection.tsx           |     100 |      100 |     100 |     100 |
 src/components/Button        |     100 |      100 |     100 |     100 |
  index.tsx                   |     100 |      100 |     100 |     100 |
 src/components/CreateNew     |     100 |      100 |     100 |     100 |
  index.tsx                   |     100 |      100 |     100 |     100 |
 src/components/Dashboard     |     100 |      100 |     100 |     100 |
  NoUrlFound.tsx              |     100 |      100 |     100 |     100 |
  UrlList.tsx                 |     100 |      100 |     100 |     100 |
  UrlListItem.tsx             |     100 |      100 |     100 |     100 |
 src/components/Footer        |     100 |      100 |     100 |     100 |
  index.tsx                   |     100 |      100 |     100 |     100 |
 src/components/Layout        |     100 |      100 |     100 |     100 |
  index.tsx                   |     100 |      100 |     100 |     100 |
 src/components/LoginModal    |     100 |      100 |     100 |     100 |
  index.tsx                   |     100 |      100 |     100 |     100 |
 src/components/Navbar        |   90.69 |       75 |      70 |      90 |
  NavbarMenuItems.tsx         |     100 |      100 |     100 |     100 |
  UserProfileButton.tsx       |     100 |      100 |     100 |     100 |
  index.tsx                   |   86.66 |    57.14 |   57.14 |    86.2 | 20,42-43,71
 src/components/ProfileIcon   |     100 |      100 |     100 |     100 |
  ProfileIcon.tsx             |     100 |      100 |     100 |     100 |
 src/components/Redirect      |     100 |      100 |     100 |     100 |
  ErrorPage.tsx               |     100 |      100 |     100 |     100 |
  LoaderTimer.tsx             |     100 |      100 |     100 |     100 |
  NotFound.tsx                |     100 |      100 |     100 |     100 |
  RedirectFooter.tsx          |     100 |      100 |     100 |     100 |
 src/components/ShimmerEffect |     100 |      100 |     100 |     100 |
  DashboardShimmer.tsx        |     100 |      100 |     100 |     100 |
  OutputSectionShimmer.tsx    |     100 |      100 |     100 |     100 |
  RedirectShimmer.tsx         |     100 |      100 |     100 |     100 |
  UserLoginShimmer.tsx        |     100 |      100 |     100 |     100 |
 src/components/Toast         |     100 |      100 |     100 |     100 |
  index.tsx                   |     100 |      100 |     100 |     100 |
 src/components/icons         |     100 |      100 |     100 |     100 |
  redirect.tsx                |     100 |      100 |     100 |     100 |
  signWithGoogle.tsx          |     100 |      100 |     100 |     100 |
 src/constants                |   89.47 |      100 |     100 |     100 |
  constants.ts                |   83.33 |      100 |     100 |     100 |
  url.ts                      |    92.3 |      100 |     100 |     100 |
 src/hooks                    |   96.42 |    66.66 |    90.9 |     100 |
  useAuthenticated.ts         |     100 |      100 |     100 |     100 |
  useToast.ts                 |   95.23 |        0 |      90 |     100 | 10
 src/pages/[redirect]         |   86.11 |       75 |   66.66 |   90.32 |
  index.tsx                   |   86.11 |       75 |   66.66 |   90.32 | 31-32,38
 src/pages/app                |     100 |      100 |     100 |     100 |
  index.tsx                   |     100 |      100 |     100 |     100 |
 src/pages/dashboard          |     100 |      100 |     100 |     100 |
  index.tsx                   |     100 |      100 |     100 |     100 |
 src/services                 |     100 |      100 |     100 |     100 |
  api.tsx                     |     100 |      100 |     100 |     100 |
 src/utils                    |     100 |       90 |     100 |     100 |
  formatDate.ts               |     100 |    88.88 |     100 |     100 | 7
  validateUrl.ts              |     100 |      100 |     100 |     100 |
------------------------------|---------|----------|---------|---------|-------------------

### Additional Notes

- The changes have been tested locally during development.
- This PR includes an extra improvement: Updated the `formatDate` utility function tests to enhance reliability and accuracy in date formatting.

### CODE REVIEW

File: src/components/CreateNew/index.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        return null;
+    }
+    return (
+        <Link
+            className="fixed bottom-10 right-4 w-12 h-12 hover:w-auto p-3 bg-blue-500 text-white cursor-pointer rounded-full hover:duration-[0.3s] active:translate-x-0.5 active:translate-y-0.5 flex items-center justify-center group flex-row hover:gap-2"
+            href="/"
+            data-testid="create-new-link"
+        >
+            <FaPlus size={20} />
+            <span className=" w-[0%] opacity-0 group-hover:w-[100%] group-hover:opacity-100  ">Create New</span>
```
Comment (ajoykumardas12) [inline src/components/CreateNew/index.tsx:19]: Few unnecessary spaces here.

**Review by ajoykumardas12 — COMMENTED** [review]
Unrelated to this PR but there was a flickering in the video when you clicked on sign in. Can you please look into that?

File: src/components/CreateNew/index.tsx (hunk 1 above)
Comment (sahsisunny) [reply src/components/CreateNew/index.tsx:19]: Can you tell me how can i fix this issue automatically?

Comment (sahsisunny) [reply src/components/CreateNew/index.tsx]: removed extra spaces

Comment (ajoykumardas12) [reply src/components/CreateNew/index.tsx]: Didn't find anything able to automate this :(

**Review by ajoykumardas12 — APPROVED** [review]

**Pull request merged**